_posts/2016-08-14-three-pillars-of-continuous-delivery.md

```diff
@@ -0,0 +1,47 @@
+---
+layout: post
+asset-type: post
+name: Three-pillars-of-Continuous-Delivery
+title: 'Three pillars of Continuous Delivery'
+date: 2016-08-14 00:00:00 +00:00
+author: Robert Firek
+image:
+    src: /assets/img/custom/blog/pillars.jpg
+    attribution:
+      text: Untitled by Donovan Christian (CC)
+      href: https://www.flickr.com/photos/donovanchristian/6208220715/
+tags:
+- continuous delivery
+- automation
+
+---
+
+Some time ago I was invited to take a part in a online panel about continuous delivery. ["Continuous Discussions"](http://electric-cloud.com/lp/continuous-discussions/) is series of one-hour online discussions between people from different backgrounds about various aspects of continuous delivery. The topic of my panel was [Deployment Automation 2.0](http://electric-cloud.com/blog/2016/07/continuous-discussions-c9d9-podcast-episode-46-deployment-automation-2-0/) where we spoke about patterns in the automation of deployments and possible future changes in this area.
+
+At the beginning we were asked to define "Must Have's" of any deployment automation. When I was looking for these "Must Have's" I tried to identify the elements which were the most important for me during the introduction of continuous delivery. What struck me the most was taking so many things required by this process for granted. Tools or the readiness of your organisation for automation were not always in my mind when I started automate "things" in different projects. This preparation gave me the possibility to identify my three pillars of good continuous delivery.
+
+### Automation, Automation, Automation
+
+**Automate the right thing.** It sounds very easy, but when I look now at some my past decisions I was not always automating the right part of my process. If you do your manual steps frequently enough or doing your manual tasks take a lot of time or your process is repeatable and complex, you have a good candidate for automation. Any automation without good context might lead you to improvement of unnecessary parts of your software delivery.
```

Comment:
...parts of your software delivery

better to say

...parts of your delivery process

or

...parts of your process

Fits the context better.